service/history/task/cross_cluster_task_processor.go

```diff
@@ -89,15 +90,19 @@ func NewCrossClusterTaskProcessors(
 	options *CrossClusterTaskProcessorOptions,
 ) common.Daemon {
 	processors := make(crossClusterTaskProcessors, 0, len(taskFetchers))
-	for _, fetcher := range taskFetchers {
-		processor := newCrossClusterTaskProcessor(
-			shard,
-			taskProcessor,
-			fetcher,
-			options,
-		)
-		processors = append(processors, processor)
+
```

Comment:
Maybe this check could be pushed higher by the stack. There is no point of creating CrossClusterTaskProcessors. It is misleading that you call a function that does nothing.

Reply:
I considered that, open to changing it, it was just following the established pattern in the history setup where all the functions that are being created on startup are quite simple and don't have if/else toggles in the megafunction `NewEngineWithShardContext`. I don't think it significantly matters either way

Reply:
I guess it is ok. Though, I've looked into NewEngineWithShardContext and that is a mess of a function :(

Reply:
haha yeah :/ there's worse sadly too (frontend handler :/ )